lib/stack_master/template_compiler.rb

```diff
@@ -1,8 +1,11 @@
 module StackMaster
   class TemplateCompiler
+    TemplateCompilationFailed = Class.new(RuntimeError)
 
     def self.compile(config, template_file_path)
       template_compiler_for_file(template_file_path, config).compile(template_file_path)
+    rescue StandardError
```

Comment:
Yeah we could use either `#inspect`, but I think that looks too low level (use of angled brackets). It would be better to build our own error message based on the properties of the error. `e.class.name` + `e.message` could be enough (or using `e.cause` if it's not nil)?